Allow getting data from multiple tags

ivangonzalezacuna opened the pull request:
With this feature, now it's possible to fetch the alerts and dashboards for multiple tags.

This feature was also needed in our company, and it was rather simple to implement.

The idea is just adding a list of tags separated by commas. So instead of `my-tag`, now you can use `my-tag-1,my-tag-2`, and the API will make 2 requests to fetch all the information for both tags. Finally, I'm executing an extra check to remove duplicates, as a dashboard could have multiple tags and be fetched multiple times.

This change is small and should be easy to review, but if you need more data, please don't hesitate asking.

heyLu commented on src/api.ts:
```diff
@@ -70,30 +70,56 @@ export class GrafanaApiClient implements GrafanaApi {
+    const alertNames = alerts.map(a => a.name);
+    return alerts.filter(
+      // Checks if the alertName is present more than once, so the `index + 1` is required to avoid finding itself
+      ({ name }, index) => !alertNames.includes(name, index + 1),
```
Same here, should probably filter by id.

ivangonzalezacuna commented:
Applied the requested changes. I added a new parameter to the dashboard, so the ID is also stored in there and the filtering can be performed.

ivangonzalezacuna commented:
Hey @K-Phoen , any update on this? We need this feature and the other I created so we can integrate the plugin in our project :)

heyLu reviewed (approved):
LGTM.

heyLu commented:
Sorry to ping again @K-Phoen, could you have a look at this MR?  We'd really like to switch back to this plugin again, so far we have to use a fork.

arnouthoebreckx commented:
@K-Phoen @dweber019  @jmfrancois  Any way to review this, our organization could use the same feature :)

dweber019 commented on src/api.ts:
```diff
@@ -70,30 +70,57 @@ export class GrafanaApiClient implements GrafanaApi {
   }
 
   async dashboardsByTag(tag: string): Promise<Dashboard[]> {
```
Probably would call it `tags` as there could be many.

ivangonzalezacuna replied:
I have rebased the changes from main and re-done the changes. I used now `tags` as it makes more sense. Good catch!

dweber019 commented on src/api.ts:
```diff
@@ -70,30 +70,57 @@ export class GrafanaApiClient implements GrafanaApi {
+    );
   }
 
   async alertsByDashboardTag(tag: string): Promise<Alert[]> {
```
Probably would call it `tags` as there could be many.

dweber019 reviewed (approved):
Else I think it's good but not my call.

ivangonzalezacuna commented:
Hi @K-Phoen , I have updated the PR to have the latest changes from the main branch. I have also simplified a bit how to do it. I tested it locally and it's working fine for both clients.

It would be great if this gets merged, then we could use this plugin instead our forked version, because we need the multiple-tag-approach for our use-case.

arnouthoebreckx commented:
@K-Phoen Possible to get a review on this?

jobosk commented:
Not to step on anyone's toes, but I've also forked K-Phoen's repo and implemented [my own version of this](https://github.com/K-Phoen/backstage-plugin-grafana/pull/61) that supports several logical operators and doesn't require an additional annotation.

K-Phoen commented:
This should be addressed by #63 :)
